Add setting Force GPU as render device

pdamasceno opened the pull request:
- Exposed ForceGPUAsRenderDevice setting under Advanced Settings/Video
- Changed the export of NvOptimusEnablement to be based on the value of ForceGPUAsRenderDevice stored in the basic.ini file

prsdamasceno committed:
Add DisableGPUAsRenderDevice setting

nodeobs_settings.cpp
- Added DisableGPUAsRenderDevice to advanced settings under video
subcategory

nodeobs_api.cpp
- Added DisableGPUAsRenderDevice default value to basic.ini config file

prsdamasceno committed:
Change export of NvOptimusEnablement

Now it is based on the value of DisableGPUAsRenderDevice setting on
basic.ini config file

CMakeLists.txt
- Commented flags NOWINMESSAGES and NOCTLMGR to prevent errors that
appeared after using #if defined(_WIN32) in the client side

server and client main.cpp
- Added a lambda function to export NvOptimisEnablement with the right
value based on DisableGPUAsRenderDevice value

prsdamasceno committed:
Add push back of DisableGPUAsRenderDevice to entries vector

nodeob_settings.cpp
- Added missing push_back to the entries vector. Now the
DisableGPUAsRenderDevice setting is correctly exposed

EddyGharbi commented on obs-studio-server/source/main.cpp:
```diff
@@ -53,6 +51,49 @@ extern "C" __declspec(dllexport) DWORD NvOptimusEnablement = 1;
+	}
+
+	// Enable high performance graphics rendering
+	return 1;
```
Can we make it a bit easier to read? We could initialize the `settingValue` to true and only change the it if we successfully find the config file and find the setting in it. Then return the `settingValue` in all cases.

pdamasceno replied:
I did the changes you asked and ended up changing the name of the setting to be able to return settingValue directly and don't make everything confusing. I believe it is better now.

prsdamasceno committed:
Change the name of the the setting to EnableGPUAsRenderDevice

The previous name of the setting was not in line with the changes asked
in the pull request.

server/client main.cpp
- Updated the lambda function that exports NVidiaOptimusEnablement

nodeobs_configManager.cpp
- Added the setting of the default value of EnableGPuAsRenderDevice
setting, after the rebase the related changes done in a previous commit
went missing because the setting of the default values where moved to
nodeobs_configManager from nodeobs_api

nodeobs_settings.cpp
- Updated the name of the setting

prsdamasceno committed:
Change the name of the setting to ForceGPUAsRenderDevice

client and server main.cpp
* Added a break after checking the value of ForceGPUAsRenderDevice to
prevent looping through the whole config file if the value is true

EddyGharbi commented on obs-studio-server/source/nodeobs_configManager.cpp:
```diff
@@ -172,6 +172,11 @@ void initBasicDefault(config_t* config)
 
+	// Set default value for EnableGPUAsRenderDevice
+	if (!config_has_user_value(config, "Video", "ForceGPUAsRenderDevice")) {
+		config_set_bool(config, "Video", "ForceGPUAsRenderDevice", true);
```
Sorry I miss this during my first review but I would use `config_set_default_bool` here in any cases, this is what it is meant for.

pdamasceno replied:
I did the change and tested everything. It is working successfully.

prsdamasceno committed:
Change set default value of ForceGPUAsRenderDevice to use config_set_default
